tests/test_rate_limiter.py

```
@@ -0,0 +1,293 @@
+
+        await limiter.shutdown()
+
+    def test_get_status_no_data(self, limiter):
+        """Test status for unknown endpoint."""
+        status = limiter.get_status("unknown")
+        assert status["endpoint"] == "unknown"
+        assert status["status"] == "no_data"
+
+    def test_get_status_specific_endpoint(self, limiter):
+        """Test status for specific endpoint."""
+        bucket = limiter._get_bucket("endpoint")
```

Ruff (0.15.2)

[error] Local variable `bucket` is assigned to but never used

Remove assignment to unused variable `bucket`

(F841)